regparser/notice/compiler.py

```diff
@@ -258,27 +258,29 @@ def replace_node_and_subtree(self, node):
         else:
             parent = self.get_parent(node)
 
-        prev_idx = [idx for idx, c in enumerate(parent.children)
-                    if c.label == node.label]
-        if prev_idx:
-            # replace existing element in place
-            prev_idx = prev_idx[0]
-            parent.children = (parent.children[:prev_idx] + [node] +
-                               parent.children[prev_idx + 1:])
-        else:
-            # actually adding a new element
-            parent.children = self.add_child(parent.children, node,
-                                             getattr(parent, 'child_labels',
-                                                     []))
-
-        # Finally, we see if this node is the parent of any 'kept' children.
-        # If so, add them back
-        label_id = node.label_id()
-        if label_id in self._kept__by_parent:
-            for kept in self._kept__by_parent[label_id]:
-                node.children = self.add_child(node.children, kept,
-                                               getattr(node, 'child_labels',
-                                                       []))
+        # not sure what to do if parent is null....
```

Comment:
Yes, I haven't fixed this yet. The bit that was causing the problem was in
Reg Z, which is why I avoided working on it for now. I'll probably just
take this out until I can figure out where exactly things went south.

On Tue, Mar 17, 2015 at 7:37 PM, CM Lubinski <notifications@github.com>
wrote:

> In regparser/notice/compiler.py
> <https://github.com/cfpb/regulations-parser/pull/264#discussion_r26627269>
> :
>
> > -                               parent.children[prev_idx + 1:])
> > -        else:
> > -            # actually adding a new element
> > -            parent.children = self.add_child(parent.children, node,
> > -                                             getattr(parent, 'child_labels',
> > -                                                     []))
> > -
> > -        # Finally, we see if this node is the parent of any 'kept' children.
> > -        # If so, add them back
> > -        label_id = node.label_id()
> > -        if label_id in self._kept__by_parent:
> > -            for kept in self._kept__by_parent[label_id]:
> > -                node.children = self.add_child(node.children, kept,
> > -                                               getattr(node, 'child_labels',
> > -                                                       []))
> > +        # not sure what to do if parent is null....
>
> This is still pretty serious. If this is needed, it means that a
> paragraph/section of the reg was supposed to be modified but was not. Which
> rule is causing the problem?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cfpb/regulations-parser/pull/264/files#r26627269>.
>



-- 
http://www.google.com/profiles/grapesmoker